rdkit/Chem/MolKey/MolKey.py

```diff
@@ -410,7 +412,8 @@ def GetKeyForCTAB(ctab, stereo_info=None, stereo_comment=None, logger=None):
   try:
     err, inchi, fixed_mol = GetInchiForCTAB(ctab)
   except BadMoleculeException:
-    logger.warn('Corrupt molecule substituting no-struct: --->\n{0}\n<----'.format(ctab))
+    msg = u'Corrupt molecule substituting no-struct: --->\n{0}\n<----'.format(ctab)
```

Comment:
That one drove me mad. Without the `u`, the (new) test that raises the Exception failed with a unicode error on Python 2.7. This works, but I honestly don't understand it why it was necessary.